domains/certificates/RegistryConnector/Worker/EventHandlers/CertificateMarkedAsIssuedEventHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using DataContext;
+using MassTransit;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ProjectOriginClients.Models;
+
+namespace RegistryConnector.Worker.EventHandlers;
+public record CertificateMarkedAsIssuedEvent
+{
+    public required string WalletUrl { get; init; }
+    public required byte[] WalletPublicKey { get; init; }
+    public required uint WalletEndpointPosition { get; init; }
+    public required string Registry { get; init; }
+    public required Guid CertificateId { get; init; }
+    public required uint Quantity { get; init; }
+    public required byte[] RandomR { get; init; }
+}
+
+public class CertificateMarkedAsIssuedEventHandler : IConsumer<CertificateMarkedAsIssuedEvent>
+{
+    private readonly IHttpClientFactory httpClientFactory;
+    private readonly ILogger<CertificateMarkedAsIssuedEventHandler> logger;
+
+    public CertificateMarkedAsIssuedEventHandler(IHttpClientFactory httpClientFactory, ILogger<CertificateMarkedAsIssuedEventHandler> logger)
+    {
+        this.httpClientFactory = httpClientFactory;
+        this.logger = logger;
+    }
+
+    public async Task Consume(ConsumeContext<CertificateMarkedAsIssuedEvent> context)
+    {
+        var message = context.Message;
+        logger.LogInformation("Sending slice to Wallet with url {WalletUrl} for certificate id {certificateId}.",
+            message.WalletUrl, message.CertificateId);
+
+        var request = new ReceiveRequest()
+        {
+            CertificateId = new FederatedStreamId()
+            {
+                Registry = message.Registry,
+                StreamId = message.CertificateId,
+            },
+            Position = message.WalletEndpointPosition,
+            PublicKey = message.WalletPublicKey,
+            Quantity = message.Quantity,
+            RandomR = message.RandomR,
+            HashedAttributes = new List<HashedAttribute>()
+        };
+
+        using var client = httpClientFactory.CreateClient();
+        var requestStr = JsonSerializer.Serialize(request);
+        var content = new StringContent(requestStr, Encoding.UTF8, "application/json");
+
+        // The Wallet is idempotent wrt. sending the same ReceiveRequest.
+        var res = await client.PostAsync(message.WalletUrl, content);
+        res.EnsureSuccessStatusCode();
```

Comment:
what happens if we dont get a successStatusCode, do we break?

Reply:
Yes - then an exception is thrown and retry takes over